app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageViewModel.kt

```diff
@@ -242,13 +246,16 @@ class NewMessageViewModel @Inject constructor(
             }
         }
 
+        val defaultSignature = SignatureController.getDefaultSignature(realm, draftMode)
         val shouldPreselectSignature = draftMode == DraftMode.REPLY || draftMode == DraftMode.REPLY_ALL
         val signature = if (shouldPreselectSignature) {
-            guessMostFittingSignature(previousMessage!!, signatures)
+            defaultSignature ?: guessMostFittingSignature(previousMessage!!, signatures)
         } else {
-            SignatureController.getSignature(realm)
+            defaultSignature
+        }
+        (signature ?: SignatureController.getDefaultSignatureWithFallback(realm, draftMode))?.let {
```

Comment:
There seems to be an issue somewhere around here.
When the default signature is set to "none", the UI fallback to another one.